### DISCUSSION TITLE
Accessing data behind a row id (_id)

### DISCUSSION THREAD

**Discussion by Serisar** [Budibase/budibase#13913] [Help]
Hello,

i have three different data tables (all internal budibase dbs), "Events", "Users" (not to be confused with the app users), and "Bookings". Each event can have several bookings, which each belong to a user and each user can have several bookings.

Now i want someone to be able to "plan" an event and pick the people who have shown their availability by putting in a booking for an event via an update form on a specific event row (filtering all my bookings via url.id for a single event works fine). Users can be selected via a relationship picker (I am picking a booking, which itself is linked to a user). 
I now want to access the user data or the booking data (in this case the email, originally stored in the user table, but accessible in the booking table via formula as well) from the relationship picker. When i try to access the selected bookings via {{ form.relationshipPicker }} i get a row _id for the corresponding booking, but i can't actually do anything with it. When i try something like {{ form.relationshipPicker.UserMail }} or {{ form.relationshipPicker.[0].UserMail }} i just get nothing.

From my understanding after reading [Bindings Scope](https://docs.budibase.com/docs/bindings-scope) it should be possible to somehow access the data because the _id is a unique identifier which points to a specific row and the context is exposed by a data provider for the bookings on the same screen.

**Comment by mikesealey** [comment]
Hey @Serisar 

Have you tried using nested data providers? Depending on your use-case it might be important which order you use them in, but nesting data providers inside each other, and putting the filter options to use should get you closer to what you're looking for.

<img width="1417" alt="image" src="https://github.com/Budibase/budibase/assets/39269380/799850cc-1299-4567-a88a-2eaf9a89fd74">

Meanwhile, work is taking place in the background to roll out the ability to query internal Budibase DB tables in a similar way to SQL queries, which might offer a neater solution for you. Keep a look out for discussions, releases, or updates that mention SQS.
